client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/application/GlobalApp.java

```diff
@@ -106,7 +106,7 @@ static public SessionState getSession(long instance) {
     static public void freeSession(long instance) {
         if (GlobalApp.sessionMap.containsKey(instance)) {
             GlobalApp.sessionMap.remove(instance);
-            LibFreeRDP.freeInstance(instance);
+            //LibFreeRDP.freeInstance(instance);
```

Comment:
Memory leak right here...

Reply:
I think GC in Android prevents this issue. At the same time, it solves problem, when app crashes after closing the window

Reply:
Actually no, native code is not garbage collected. It is only freed up when the application exits. As for the issue, could you open a bug report for that and give some details? Better fix it than have some hack like this.

Reply:
Issue exists only on Samsung DeX. It can be connected with JNI implementation. You can see what happens after manually closing window on screenshot attached (![screenshot_20170919-152336](https://user-images.githubusercontent.com/28042554/30592928-6d4e210a-9d51-11e7-8e4f-2b065640ebb4.png)). We are investigating this issue, but as temporary workaround I suggest Thread.sleep() in case of DeX mode.